build.rs

```diff
@@ -419,6 +423,8 @@ fn build_c_code(
     core_name_and_version: &str,
 ) {
     let (asm_srcs, obj_srcs) = if let Some(asm_target) = asm_target {
+        println!("cargo:rustc-cfg=perlasm");
```

Comment:
It seems like the `cargo:rustc-cfg` directives need to be emitted from `generate_sources_and_preassemble` instead so that the `package` job (mk/package.sh) succeeds.

Reply:
I believe that wouldn't cover the case where `RING_PREGENERATE_ASM != 1 && !is_git`. I've moved these `println!()` statements into `main()` itself, which covers all cases.